File: src/shipping/shipping.ts
export interface ShippingInput {
  country: string;
  subtotalCents: number;
  weightGrams: number;
}

export function shippingCost({
  country,
  subtotalCents,
  weightGrams,
}: ShippingInput): number {
  if (weightGrams < 0) {
    throw new RangeError("weightGrams must be >= 0");
  }
  if (country === "US") return subtotalCents >= 5000 ? 0 : 599;
  return 1200 + Math.ceil(weightGrams / 1000) * 400;
}

File: python/shipping.py
"""Chapter 10."""
import math


def shipping_cost(country: str, subtotal_cents: int, *,
                  weight_grams: int) -> int:
    if weight_grams < 0:
        raise ValueError("weight_grams must be >= 0")
    if country == "US":
        return 0 if subtotal_cents >= 5000 else 599
    return 1200 + math.ceil(weight_grams / 1000) * 400
